server/src/main/java/org/elasticsearch/transport/TransportStats.java

```diff
@@ -196,15 +185,9 @@ public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params outerP
             if (inboundHandlingTimeBucketFrequencies.length > 0) {
                 histogramToXContent(builder, inboundHandlingTimeBucketFrequencies, Fields.INBOUND_HANDLING_TIME_HISTOGRAM);
                 histogramToXContent(builder, outboundHandlingTimeBucketFrequencies, Fields.OUTBOUND_HANDLING_TIME_HISTOGRAM);
-            } else {
-                // Stats came from before v8.1
-                assert IMPOSSIBLE_IN_V9;
             }
             if (transportActionStats.isEmpty() == false) {
                 builder.startObject(Fields.ACTIONS);
-            } else {
-                // Stats came from before v8.8
-                assert IMPOSSIBLE_IN_V9;
```

Comment:
And

```java
assert transportActionStats.isEmpty() == false;
builder.startObject(Fields.ACTIONS);
```

Or just

```java
builder.startObject(Fields.ACTIONS);
```

Reply:
Good point, now we should always serialize actions.